### PR TITLE
TDL-17703 Update projects stream

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change
- Updated the endpoint for the projects stream as the one used in the tap was deprecated ([Ref Link](https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-projects/#api-rest-api-2-project-get))
- As the new endpoint supported pagination, added pagination support for the sync
- Updated integration tests for paginated sync
- Added unittests for the code change.

# Manual QA steps
 - Run the sync and check whether the records are returned in a paginated manner.
 - Check the path is changed for projects stream.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by dbshah1212 — APPROVED** [review]

File: tap_jira/streams.py [hunk 1]
```diff
@@ -115,7 +116,7 @@ def write_page(self, page):
 
 
 class Projects(Stream):
-    def sync(self):
+    def sync_on_prem(self):
```
Comment (dbshah1212) [inline tap_jira/streams.py:119]: Do add function comment and reason for this name change.

File: tap_jira/streams.py [hunk 2]
```diff
@@ -140,6 +141,48 @@ def sync(self):
                 for page in pager.pages(COMPONENTS.tap_stream_id, "GET", path):
                     COMPONENTS.write_page(page)
 
+    def sync_cloud(self):
```
Comment (dbshah1212) [inline tap_jira/streams.py:144]: Do add a function comment and reason for adding this function.

File: tap_jira/streams.py [hunk 3]
```diff
@@ -140,6 +141,48 @@ def sync(self):
...
+        offset = 0
+        while True:
+            params = {
```
Comment (dbshah1212) [inline tap_jira/streams.py:147]: If the inner code is the same then can we reuse the code of def sync_on_prem?

Comment (dbshah1212) [reply tap_jira/streams.py:147]: I can see redundant codes so we can make a seprate function as well and call after request call.

File: tap_jira/streams.py (hunk 1 above)
Comment (namrata270998) [reply tap_jira/streams.py:119]: added comments

**Review by dbshah1212 — APPROVED** [review]

**Review by kspeer825 — APPROVED** [review]

File: tap_jira/__init__.py [hunk 4]
```diff
@@ -109,15 +114,18 @@ def sync():
 def main():
     args = get_args()
 
+    jira_config = args.config
+    # jira client instance
+    jira_client = Client(jira_config)
+
```
Comment (KrisPersonal) [inline tap_jira/__init__.py:120]: Why do we need to have a separate variable called jira_config? args.config could be used to get jira_client right?

Comment (namrata270998) [reply tap_jira/__init__.py:120]: Yes, args.config can be used, we've just assigned it to a variable for reuse later.

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**